src/transform/BodySpacingTransform.ts

```diff
@@ -0,0 +1,65 @@
+interface Spacing {
+  top?: string
+  right?: string
+  bottom?: string
+  left?: string
+}
+
+/**
+ * Sets the margins on the body element via inline styles.
+ * @param {!Document} document
+ * @param {Spacing} values { top, right, bottom, left }
+ *   Use value strings with units, e.g. '16px'. Undefined values are ignored.
+ * @param callback callback function
+ * @return {void}
+ */
+const setMargins = (document: Document, values: Spacing, callback?: () => void): void => {
+  const body: HTMLElement = document.body
+  if (values.top !== undefined) {
+    body.style.marginTop = values.top
+  }
+  if (values.right !== undefined) {
+    body.style.marginRight = values.right
+  }
+  if (values.bottom !== undefined) {
+    body.style.marginBottom = values.bottom
+  }
+  if (values.left !== undefined) {
+    body.style.marginLeft = values.left
+  }
+  if (callback) {
+    callback()
```

Comment:
Wondering if we can do without the callback here and in `setPadding` (since they're not passed anything and they both happen at very end of their respective methods)? i.e. could you just directly call anything which needs to happen after `setMargins` or `setPadding` by invoking that code after calls to `setMargins` / `setPadding`?

Reply:
I suspect this is true. I mainly added it because it was requested in the ticket. @joewalsh : thoughts?

Reply:
The thought was to add a `requestAnimationFrame` before the callback to ensure the update was rendered before the callback is called - this way the apps could adjust the margins on launch and not remove the cover until after they've changed. Same with the theme and other visual methods with callbacks.

Reply:
@montehurd Any other concerns?

Reply:
@berndsi I added a couple small notes. Not must-haves but may be nice.